claim-check/code-samples/sample-2/client-consumer/EventHubsConsumer.cs

```diff
@@ -76,36 +35,87 @@ public void Configure()
-            var utcNow = DateTime.UtcNow;
-            var receivers = runtimeInfo.PartitionIds
-                .Select(pid => {
-                    var receiver = client.CreateReceiver("$Default", pid, EventPosition.FromEnqueuedTime(utcNow));
-                    Console.WriteLine("Created receiver for partition '{0}'.", pid);
-                    receiver.SetReceiveHandler(new PartitionReceiver(storageAccount, downloadDestination));
-                    return receiver;
-                })
-                .ToList();
+            async Task processEventHandlerAsync(ProcessEventArgs eventArgs)
+            {
+                if (eventArgs.CancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+                try
+                {                  
+                    string body = Encoding.UTF8.GetString(eventArgs.Data.Body.ToArray());
+                    var jsonMessage = JArray.Parse(body).First;
+                    Uri uploadedUri = new Uri(jsonMessage["data"]["url"].ToString());
+                    Console.WriteLine("Blob available at: {0}", uploadedUri);
+                    BlockBlobClient blockBlob = new BlockBlobClient(uploadedUri);
+                    string uploadedFile = Path.GetFileName(jsonMessage["data"]["url"].ToString());
+                    string destinationFile = Path.Combine(downloadDestination, Path.GetFileName(uploadedFile));
+                    Console.WriteLine("Downloading to {0}...", destinationFile);
+                    await blockBlob.DownloadToAsync(destinationFile);
+                    Console.WriteLine("Done.");
+                    Console.WriteLine();
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine($"An error was observed while processing events.  Message: { ex.Message }");
+                }             
+            };
+
+            Task processErrorHandler(ProcessErrorEventArgs eventArgs)
+            {
+                if (eventArgs.CancellationToken.IsCancellationRequested)
+                {
+                    return Task.CompletedTask;
+                }
+                Console.WriteLine();
+                Console.WriteLine("===============================");
+                Console.WriteLine($"The error handler was invoked during the operation: { eventArgs.Operation ?? "Unknown" }, for Exception: { eventArgs.Exception.Message }");
+                Console.WriteLine("===============================");
+                Console.WriteLine();
+                return Task.CompletedTask;
+            }
+            processor.PartitionInitializingAsync += partitionInitializingHandler;
+            processor.ProcessEventAsync += processEventHandlerAsync;
+            processor.ProcessErrorAsync += processErrorHandler;
 
             try
             {
+                await processor.StartProcessingAsync();
                 await Task.Delay(-1, cancellationToken);
```

Comment:
nit:  It may be worthy of a comment here to let readers know that the delay call will block until the cancellation token is signaled.   Or, perhaps, consider using `Timeout.Infinite` rather than `-1` for clarity.